pkg/loki/config_wrapper_test.go

```diff
@@ -579,6 +579,46 @@ storage_config:
 			assert.EqualValues(t, defaults.Ruler.StoreConfig.S3, config.Ruler.StoreConfig.S3)
 		})
 
+		t.Run("named storage config provided via config file is preserved", func(t *testing.T) {
+			namedStoresConfig := `common:
+  storage:
+    s3:
+      endpoint: s3://common-bucket
+      region: us-east1
+      access_key_id: abc123
+      secret_access_key: def789
+storage_config:
+  named_stores:
+    aws:
+      store-1:
+        endpoint: s3://foo-bucket
+        region: us-west1
+        access_key_id: 123abc
+        secret_access_key: 789def
+      store-2:
+        endpoint: s3://bar-bucket
+        region: us-west2
+        access_key_id: 456def
+        secret_access_key: 789abc`
+			config, _ := testContext(namedStoresConfig, nil)
+
+			// should be set by common config
```

Comment:
In this case, since we defined storage in 1) common config, and in 2) named storage, the common config will be ignored and the named storage will be the one used, right? Any way we can assert that in the test?

Reply:
common config won't be ignored as it overrides StorageConfig (if it's not configured). And users can choose to use one or both by setting objectType across periodConfigs.
`aws` for using configuration from StorageConfig (which actually comes from common config)
`aws.named-store-key-name` for using the named store

Reply:
Ah got it 👍 not necessary for this PR, but maybe it would be useful to add this information to period_config, that one can set it via the StorageConfig or via named_stored